README.md

```diff
@@ -0,0 +1,140 @@
+
+# Nuxeo Cold Storage
+
+The Nuxeo Cold Storage addon allows the storage of the document main content in a cold storage. This can be needed for archiving, compliance, etc.
+
+For more details around functionalities, requirements, installation and usage please consider this addon [official documentation](TBD).
+
+## Context
+Nuxeo Cold Storage is an addon that can be plugged to Nuxeo. 
+
+It is bundled as a marketplace package that includes all the backend and frontend contributions needed for [Nuxeo Platform](https://github.com/nuxeo/nuxeo) and [Nuxeo Web UI](https://github.com/nuxeo/nuxeo-web-ui).
+
+## Sub Modules Organization
+
+- **ci**: CI/CD files and configurations responsible to generate preview environments and running Cold Storage pipeline
+- **nuxeo-coldstorage**: Backend contribution for Nuxeo Platform
+- **nuxeo-coldstorage-package**: Builder for [nuxeo-coldstorage](https://nos-preprod-connect.nuxeocloud.com/nuxeo/site/marketplace/package/nuxeo-retention?version=2021.0.2) marketplace package. This package will install all the necessary mechanisms to integrate Cold Storage capabilities into Nuxeo
+- **nuxeo-coldstorage-web**: Frontend contribution for Nuxeo Web UI
+
+## Build
+
+Nuxeo's ecosystem is Java based and uses Maven. This addon is not an exception and can be built by simply performing:
+
+```shell script
+mvn clean install
+```
+
+This will build all the modules except _ci_ and generate the correspondent artifacts: _`.jar`_ files for the contributions, and a _`.zip_ file for the package.
+
+### Frontend Contribution
+
+`nuxeo-coldstorage-web` module is also generating a _`.jar`_ file containing all the artifacts needed for an integration with Nuxeo's ecosystem.
+Nevertheless this contribution is basically generating an ES Module ready for being integrated with Nuxeo Web UI.
+
+It is possible to isolate this part of the build by running the following command:
+
+```shell script
+npm run build
+```
+
+It is using [rollup.js](https://rollupjs.org/guide/en/) to build, optimize and minify the code, making it ready for deployment.
+
+## Test
+
+In a similar way to what was written above about the building process, it is possible to run tests against each one of the modules.
+
+Here, despite being under the same ecosystem, the contributions use different approaches.
+
+### Backend Contribution
+
+#### Unit Tests
+
+```shell script
+mvn test
+```
+
+### Frontend Contribution
+
+#### Unit Tests
+
+```shell script
+npm run test
+```
+
+[Web Test Runner](https://modern-web.dev/docs/test-runner/overview/) is the test runner used to run this contribution unit tests.
+The tests run against bundled versions of Chromium, Firefox and Webkit, using [Playwright](https://www.npmjs.com/package/playwright)
+
+#### Functional Tests
+
+```shell script
+npm run ftest
+```
+
+To run the functional tests, [Nuxeo Web UI Functional Testing Framework](https://github.com/nuxeo/nuxeo-web-ui/tree/master/packages/nuxeo-web-ui-ftest) is used.
```

Comment:
should point to the web-ui maintenance branch https://github.com/nuxeo/nuxeo-web-ui/tree/maintenance-3.0.x/packages/nuxeo-web-ui-ftest